### ISSUE TITLE
Function/expression interpolation fails with Nedelec elements (Hcurl/Hdiv)

### ISSUE THREAD

**Issue by IgorBaratta** [FEniCS/dolfinx#676]
Minimum failing example:
```
import dolfin
import numpy


mesh = dolfin.UnitCubeMesh(dolfin.MPI.comm_world, 2, 2, 2, dolfin.cpp.mesh.CellType.tetrahedron)
V = dolfin.FunctionSpace(mesh, ("N1curl", 1))
# V = dolfin.VectorFunctionSpace(mesh, ('CG', 1))


def expression(x):
    values = numpy.empty((3, x.shape[1]))
    values[0] = x[0] + x[1] + x[2]
    values[1] = x[0] - x[1] - x[2]
    values[2] = x[0] + x[1] + x[2]
    return values


u = dolfin.Function(V)
u.interpolate(expression)
```

Executing this simple script causes segmentation fault (core dumped).
Same happens with `N1div`.

**Comment by garth-wells** [comment]
This is a known problem - what we need is a way to tell from from the generated code whether or not the dofs are simple point evaluations, and if they're not throw an exception (until interpolation in H(div/curl/...) is implemented). 

Do we already have a simple and reliable way to infer the dof type?

**Comment by IgorBaratta** [comment]
I don't know how can we infer the dof type from the generated code.  Maybe from the element family, but don't think it's a good solution. 
Should this information be provided by ffc? (eg: infering from mapping type) or by fiat?

**Comment by IgorBaratta** [comment]
Not sure if it's related to this issue, but the following code for projection also fails for constants and functions from different function spaces:

```
import dolfin
import ufl
mesh = dolfin.UnitCubeMesh(dolfin.MPI.comm_world, 2, 2, 2,
                           dolfin.cpp.mesh.CellType.tetrahedron)
V = dolfin.FunctionSpace(mesh, ("N1curl", 1))
# W = dolfin.VectorFunctionSpace(mesh, ('CG', 1))

C = dolfin.Constant(mesh, [1., 1., 1.])
u = ufl.TrialFunction(V)
v = ufl.TestFunction(V)

a = ufl.inner(u, v)*ufl.dx
L = ufl.inner(C, v)*ufl.dx


M = dolfin.Function(V)
dolfin.solve(a == L, M, petsc_options={"ksp_type": "preonly",
                                       "pc_type": "lu",
                                       "pc_factor_mat_solver_type": "mumps"})
error = dolfin.fem.assemble_scalar((M - C)**2 * ufl.dx)
assert abs(error) < 1e-9
```

However if `C` is a function from `N1curl` space it works ok.

**Comment by garth-wells** [comment]
Fixed in #1798 and other PRs.
